dashboard/app/models/levels/poetry.rb

```diff
@@ -25,15 +25,35 @@
 #
 
 class Poetry < GamelabJr
-  before_save :check_default_poem
+  validate :validate_default_poem_and_dropdown_poems
 
   serialized_attrs %w(
     default_poem
     standalone_app_name
+    dropdown_poems
   )
 
-  def check_default_poem
-    self.default_poem = nil unless Poetry.subtypes_with_poems.include?(standalone_app_name)
+  def sanitize_default_poem
+    self.default_poem = nil if !Poetry.subtypes_with_poems.include?(standalone_app_name) ||
+      !Poetry.poem_keys_for_subtype(standalone_app_name).include?(default_poem)
+  end
+
+  def sanitize_dropdown_poems
```

Comment:
Q: What does sanitize mean in this context?

Reply:
We remove any poems that aren't part of the list of all poems for that subtype. I'll add a comment to make it clearer.